packages/realm/src/Dictionary.ts

```diff
@@ -209,12 +212,16 @@ export class Dictionary<T = unknown> extends Collection<
    * @since 10.5.0
    * @ts-expect-error We're exposing methods in the end-users namespace of values */
   *values(): Generator<T> {
+    const realm = this[REALM];
     const snapshot = this[INTERNAL].values.snapshot();
-    const size = snapshot.size();
+    const itemType = toItemType(snapshot.type);
+    const { fromBinding, toBinding } = this[ACCESSOR];
+    const accessor = createResultsAccessor({ realm, typeHelpers: { fromBinding, toBinding }, itemType });
+    const results = new Results<T>(realm, snapshot, accessor);
+    const size = results.length;
 
-    const { snapshotGet } = this[ACCESSOR];
     for (let i = 0; i < size; i++) {
-      yield snapshotGet(snapshot, i);
+      yield results[i];
```

Comment:
Maybe it's me, but isn't this a bit overly complicated? Can we do something like:

```ts
const realm = this[REALM];
const results = this[INTERNAL].values;
const itemType = toItemType(snapshot.type);
const { fromBinding, toBinding } = this[ACCESSOR];
const accessor = createResultsAccessor({ realm, typeHelpers: { fromBinding, toBinding }, itemType });
const results = new Results<T>(realm, snapshot, accessor);
return results.values();
```

Essentially, construct the managed results from `this[INTERNAL].values` and return its generator instead (it'll do the snapshotting already, making the original snapshot unnecessary).

Reply:
The generator function explicitly needs to `yield` the next value, returning the Results generator won't work. You'd have to replace `return results.values()` to the following:
```ts
for (const value of results.values()) {
  yield value;
}
```

Reply:
Applied ✅